Assets/Tests/InputSystem/CoreTests_GlobalActions.cs

```diff
@@ -0,0 +1,500 @@
+#if UNITY_INPUT_SYSTEM_ENABLE_GLOBAL_ACTIONS_API
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using NUnit.Framework;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.InputSystem;
+using UnityEngine.InputSystem.Editor;
+using UnityEngine.InputSystem.Interactions;
+using UnityEngine.InputSystem.Utilities;
+using UnityEngine.TestTools;
+using Input = UnityEngine.InputSystem.Input;
+
+internal partial class CoreTests
+{
+    private const string TestCategory = "GlobalActions";
+    private string m_TemplateAssetPath;
+
+    [SetUp]
+    public override void Setup()
+    {
+        // this asset takes the place of InputManager.asset for the sake of testing, as we don't really want to go changing
+        // that asset in every test.
+        var testInputManager = ScriptableObject.CreateInstance<TestInputManager>();
+        AssetDatabase.CreateAsset(testInputManager, "Assets/TestInputManager.asset");
+
+        // create a template input action asset from which the input action asset stuffed inside the InputManager will be created
+        var globalActions = ScriptableObject.CreateInstance<InputActionAsset>();
+        globalActions.AddActionMap("ActionMapOne").AddAction("ActionOne");
+        m_TemplateAssetPath = Path.Combine(Environment.CurrentDirectory, "Assets/TestGlobalActions.inputactions");
+        File.WriteAllText(m_TemplateAssetPath, globalActions.ToJson());
+
+        InputSystem.SetGlobalActionAssetPaths(m_TemplateAssetPath, "Assets/TestInputManager.asset");
+
+        base.Setup();
+    }
+
+    [TearDown]
+    public override void TearDown()
+    {
+        if (File.Exists(m_TemplateAssetPath))
+            File.Delete(m_TemplateAssetPath);
+
+        AssetDatabase.DeleteAsset("Assets/TestInputManager.asset");
+
+        base.TearDown();
+    }
+
+    [Test]
+    [Category(TestCategory)]
+    public void GlobalActions_TemplateAssetIsInstalledOnFirstUse()
+    {
+        var asset = GlobalActionsAsset.GetOrCreateGlobalActionsAsset("Assets/TestInputManager.asset", m_TemplateAssetPath);
+
+        Assert.That(asset, Is.Not.Null);
+        Assert.That(asset.actionMaps.Count, Is.EqualTo(1));
+        Assert.That(asset.actionMaps[0].actions.Count, Is.EqualTo(1));
+    }
+
+    [Test]
+    [Category(TestCategory)]
+    public void GlobalActions_CanQueryActionsByStringName()
+    {
+        var keyboard = InputSystem.AddDevice<Keyboard>();
+
+        InputSystem.actions.FindAction("ActionOne").AddBinding("<keyboard>/w");
+
+        Set(keyboard.wKey, 1);
+
+        Assert.That(Input.WasPressedThisFrame("ActionOne"), Is.True);
```

Comment:
As we are proposing removing these Input.IsPressed/WasPRessetThisFrame/WasReleasedThisFrame functions then this test should be removed too.